cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandler.java

```diff
@@ -209,11 +232,19 @@ public void executeAdmin(HttpRequest request, HttpResponder responder, @PathPara
   @POST
   @Path("/data/datasets/{name}/data/{method}")
   public void executeDataOp(HttpRequest request, HttpResponder responder, @PathParam("namespace-id") String namespaceId,
-                            @PathParam("name") String name, @PathParam("method") String method) {
+                            @PathParam("name") String name, @PathParam("method") String method) throws Exception {
+
+    Id.Namespace namespace = Id.Namespace.from(namespaceId);
+    ensureNamespaceExists(namespace);
+
     // todo: execute data operation
     responder.sendStatus(HttpResponseStatus.NOT_IMPLEMENTED);
   }
 
+  private void ensureNamespaceExists(Id.Namespace namespace) throws Exception {
+    namespaceAdmin.get(namespace);
```

Comment:
you will need special handling for ``Id.Namespace.SYSTEM``, like there was in ``DatasetInstanceService``